docs/guide/docs/tutorials/chat-3rd-party-OAuth.md

```diff
@@ -0,0 +1,139 @@
+
+The code for this example is available in the [examples](https://github.com/botpress/botpress/tree/master/examples/chat-3rd-party-OAuth) directory of our GitHub repository.
+
+## Register the module
+
+Botpress modules provide a simple yet powerful way extend your bot capabilities without altering the Core. If we think about our use case here, the only thing that our module has to do here is to offer an api so we can perform the steps to authenticate the user to Twitter and only that. No UI, no middlewares, only an api. First of create an `src` directory with 2 sub dirs `backend` and `views`. Let's go ahead and write a simple module entry point (backend/index.ts).
+
+```ts
+import * as sdk from 'botpress/sdk'
+
+const onServerStarted = async (bp: typeof sdk) => {}
+
+const onServerReady = async (bp: typeof sdk) => {}
+
+const onBotMount = async (bp: typeof sdk, botId: string) => {}
+
+const entryPoint: sdk.ModuleEntryPoint = {
+  onServerStarted,
+  onServerReady,
+  onBotMount,
+  definition: {
+    name: 'twitter-auth',
+    fullName: 'Twitter oAuth1 example',
+    homepage: 'https://botpress.io',
+    noInterface: true
+  }
+}
+
+export default entryPoint
+```
+
+Now that we have the shell of our module setup, we need to activate it in our `botpress.config.json` so the botpress module builder can easily find and build it. Depending on where you created your module.
+
+```js
+{
+  // MODULES_ROOTS is the modules directory, you can yse your own location
+  "location": "MODULES_ROOT/twitter-auth",
+  "enabled": true
+}
+```
+
+Now that we have a working module let's get into the implementation. For simplycity we will use Passport.js along with a Twitter Strategy to implement the OAuth flow. Go ahead an install those deps. Once that done, we will need to add your Twitter App api keys so that our bot can properly authenticate your user to Twitter. That can be done by creating a module config file for each bot. Config here is pretty simple, obviously you can customize it for your needs. To enable a Twitter authentication, add the following file in the config directory of the desired bots:
+
+```js
+// bots/yourbot/config/twitter-auth.json
+{
+  "enabled": true,
+  "apiKey": "YOUR_TWITTER_APP_API_KEY"
+  "apiSecret": "YOUR_TWITTER_APP_SECRET_KEY"
+}
+```
+
+Next step, is to implement the `onBotMount` function the in our `module/src/index.ts` that will configure the Passport Twitter straregy for each bot.
+
+```ts
+const onBotMount = async (bp: typeof sdk, botId) => {
+  const modConfig = (await bp.config.getModuleConfigForBot('twitter-auth', botId)) as Config
+  if (!modConfig.enabled) {
+    return
+  }
+
+  const bpConfig = await bp.config.getBotpressConfig()
+  const baseCallbackUrl = bpConfig.externalUrl || `http://${bpConfig.host}:${bpConfig.port}`
```

Comment:
process.EXTERNAL_URL is set to the real external url. Also, if the port in configuration is already in use, the correct one is in the process environment var.